### PR TITLE
Extract fields by names in CheckpointEntryIterator

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Extract fields by names in CheckpointEntryIterator

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

**Comment by ebyhr** [comment]
Note: Fix the following failure:
```
2023-11-14T05:46:09.1553670Z [ERROR] Failures: 
2023-11-14T05:46:09.1555376Z [ERROR]   TestDeltaLakeMinioAndHmsConnectorSmokeTest>BaseDeltaLakeConnectorSmokeTest.testCheckpointing:1336 
```

**Comment by ebyhr** [comment]
CI hit #19747

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointEntryIterator.java [hunk 1]
```diff
@@ -707,10 +621,12 @@ private void fillNextEntries()
 
             // process page
             for (int i = 0; i < extractors.size(); ++i) {
-                DeltaLakeTransactionLogEntry entry = extractors.get(i).getEntry(session, page.getBlock(i).getLoadedBlock(), pagePosition);
-                if (entry != null) {
-                    nextEntries.add(entry);
```
Comment (ebyhr) [inline plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointEntryIterator.java:712]: We may want to restore this condition after #19588 + early exit in `buildAddEntry` method.

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointEntryIterator.java [hunk 2]
```diff
@@ -707,10 +621,12 @@ private void fillNextEntries()
...
+                Block block = page.getBlock(i).getLoadedBlock();
+                if (block.isNull(pagePosition)) {
+                    continue;
                 }
+                DeltaLakeTransactionLogEntry entry = extractors.get(i).getEntry(session, (RowType) columnFields.get(i).getType(), block, pagePosition);
```
Comment (ebyhr) [inline plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointEntryIterator.java:628]: This implementation has a bug when the requested field doesn't exist in Parquet file. Fixing the issue now.

**Comment by ebyhr** [comment]
CI hit #16315

**Review by findepi — APPROVED** [review]
skimmed, lgtm

awesome change, thank you!

**Pull request merged**